flink-table/flink-table-planner/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -0,0 +1,315 @@
+	}
+
+	/**
+	 * Sets the current default catalog name that will be used when resolving table path.
+	 *
+	 * @param catalogName catalog name to set as current default catalog
+	 * @throws CatalogNotExistException thrown if the catalog doesn't exist
+	 * @see CatalogManager#resolveTable(String...)
+	 */
+	public void setCurrentCatalog(String catalogName) throws CatalogNotExistException {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(catalogName), "Catalog name cannot be null or empty.");
+
+		if (externalCatalogs.keySet().contains(catalogName)) {
+			throw new CatalogException("An external catalog cannot be set as the default one.");
+		}
+
+		Catalog potentialCurrentCatalog = catalogs.get(catalogName);
+		if (potentialCurrentCatalog == null) {
+			throw new CatalogNotExistException(catalogName);
+		}
+
+		if (!currentCatalogName.equals(catalogName)) {
+			currentCatalogName = catalogName;
+			currentDatabaseName = potentialCurrentCatalog.getCurrentDatabase();
+
+			LOG.info(
+				"Sets the current default catalog as '{}' and the current default database as '{}'",
+				currentCatalogName,
+				currentDatabaseName);
+		}
+	}
+
+	/**
+	 * Gets the current default database name that will be used when resolving table path.
+	 *
+	 * @return the current default database
+	 * @see CatalogManager#resolveTable(String...)
+	 */
+	public String getCurrentDatabaseName() {
+		return currentDatabaseName;
+	}
+
+	/**
+	 * Sets the current default catalog name that will be used when resolving table path.
+	 * The database has to exist in the current catalog.
+	 *
+	 * @param databaseName database name to set as current default database name
+	 * @throws DatabaseNotExistException thrown if the database doesn't exist in the current catalog
+	 * @see CatalogManager#resolveTable(String...)
+	 * @see CatalogManager#setCurrentCatalog(String)
+	 */
+	public void setCurrentDatabase(String databaseName) throws DatabaseNotExistException {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(databaseName), "The database name cannot be null or empty");
+
+		if (!catalogs.get(currentCatalogName).databaseExists(databaseName)) {
+			throw new DatabaseNotExistException(currentCatalogName, databaseName);
+		}
+
+		if (!currentDatabaseName.equals(databaseName)) {
+			currentDatabaseName = databaseName;
+
+			LOG.info(
+				"Sets the current default catalog as '{}' and the current default database as '{}'",
+				currentCatalogName,
+				currentDatabaseName);
+		}
+	}
+
+	/**
+	 * Tries to resolve a table path to a {@link CatalogTableOperation}. First it tries to look for
+	 * {@code [default-path].[table-path]} if no table is found assumes the table path is a fully qualified one
+	 * and looks for {@code [table-path]}.
+	 *
+	 * @param tablePath table path to look for
+	 * @return {@link CatalogTableOperation} containing both fully qualified table identifier and its
+	 * {@link TableSchema}.
+	 */
+	public Optional<CatalogTableOperation> resolveTable(String... tablePath) {
```

Comment:
This method seems resolving tables registered either in Catalogs or ExternalCatalogs. What about those registered in neither of them, such as those registered by registerTable()?

Reply:
As Timo said, all objects in the end should be registered in one of the Catalogs. Right now we maintain the `ExternalCatalogs` just for backwards compatibility.

This method is used whenever we have to look up any object exclusively in the API module. This means e.g. in the `TableEnvironment#scan` method.